qiita_db/processing_job.py

```diff
@@ -703,6 +703,101 @@ def status(self):
             qdb.sql_connection.TRN.add(sql, [self.id])
             return qdb.sql_connection.TRN.execute_fetchlast()
 
+    def _generate_notification_message(self, value, error_msg):
+        ignored_software = ('artifact definition',)
+        ignored_commands = ('Validate', 'complete_job', 'release_validators')
+
+        # abort early conditions (don't send an email notification)
+        # tentatively accept the overhead of a function-call, even when a
+        # notification isn't sent, just to keep the logic clean and
+        # centralized.
+
+        if value == 'waiting':
+            # notification not needed.
+            return
+
+        if not self.user.info['receive_processing_job_emails']:
+            # notification not needed.
+            return
+
+        if self.command.software.name in ignored_software:
+            # notification not needed.
+            return
+
+        if self.command.name in ignored_commands:
+            # notification not needed.
+            return
```

Comment:
Why not return None explicitly (for reference: https://realpython.com/python-return-statement/#returning-none-explicitly)?

Reply:
I'm agnostic on this. Personally I see the value in being explicit but I understand pythonistas might find it redundant. I'll make the change.